posts/mathlib-changelog.md

```diff
@@ -0,0 +1,50 @@
+---
+author: 'David Chanin'
+category: 'Community projects'
+date: 2022-07-27 09:35:23 UTC+02:00
+description: ''
+has_math: false
+link: ''
+slug: mathlib-changelog
+tags: ''
+title: Introducing Mathlib Changelog
+type: text
+---
+
+Tldr; check out [mathlib-changelog.org](https://mathlib-changelog.org) to explore the historical changes to Mathlib, and find out what happened to that lemma you were using.
+
+
+## A Changelog for Mathlib
+As a beginner in the world of Lean and Mathlib (and formal math in general), I found that when I try to follow along with tutorials I find online that I’d often get errors when typing verbatim what I would see on the screen. Searching on Google wouldn’t help, and I often couldn’t find any reference to some of the definitions and lemmas I was seeing in the Mathlib docs.
+
+Disoriented, I eventually stumbled onto the amazing [Leanprover Community Zulip](https://leanprover.zulipchat.com/) chat where I learned that Mathlib changes very fast, so it’s common for lemmas and theorems to be added and removed all the time. This means Mathlib can improve at lightning speed, but also means that tutorials created even a few months ago likely won’t work anymore. After discussing on Zulip the pros and cons of ideas for versioning or maintaining an official changelog, it seemed like the most practical path forward would be to build a tool to automatically extract a searchable changelog from the Mathlib git history. Or it could be because I work as a software engineer, so somehow my solution to every problem is to build more software...
```

Comment:
```suggestion
Disoriented, I eventually stumbled onto the amazing [Leanprover Community Zulip](https://leanprover.zulipchat.com/) chat where I learned that Mathlib changes very fast, so it’s common for lemmas and theorems to be added and removed all the time. This means Mathlib can improve at lightning speed, but also means that tutorials created even a few months ago likely won’t work anymore, unless they come packaged as a Lean project fixing the mathlib version. After discussing on Zulip the pros and cons of ideas for versioning or maintaining an official changelog, it seemed like the most practical path forward would be to build a tool to automatically extract a searchable changelog from the Mathlib git history. Or it could be because I work as a software engineer, so somehow my solution to every problem is to build more software...
```
Your point is well taken! But I think it's a little misleading phrased like this. Tutorials like the [tutorials project](https://github.com/leanprover-community/tutorials) are always usable "out of the box," and a lot of people put in effort to keep them up to date.

Reply:
That's a good point,  I'll reword this in the next draft. Even when stuff is in a packaged Lean project that properly fixes the mathlib version, It's still confusing since I'll try to copy what I see in those projects into my own work and find I can't get it to work, often due to different mathlib versions. I'll try to explain this better, don't want to minimize the hard work of everyone keeping the tutorials up to date!